app/event_logger.rb

```diff
@@ -13,9 +14,11 @@ def self.log_command(command_class, payload, &_block)
           request_id: GdsApi::GovukHeaders.headers[:govuk_request_id]
         )
 
-        response = yield(event) if block_given?
+        response, callbacks = yield(event) if block_given?
```

Comment:
I was thinking about an alternative interface where `BaseCommand` has a `call_after_transaction_commits` method (or similar) that all of the commands inherit. The `self.call` method on `BaseCommand` could then play through those callbacks after its sure the event has been logged (and the transaction committed).

As it stands, it's not immediately obvious why those calls happen afterwards. Maybe this alternative might make the code more intention revealing? What do you think?